js/app/interfaces/card.js

```diff
@@ -157,6 +160,18 @@ const Card = new Lang.Interface({
             GObject.ParamFlags.READWRITE,
             ''),
         /**
+         * Property: collection-name
+         * The name of whatever it is a set card contains
+         *
+         * For sets, we want to show how many items are contained within it.
+         * To do this we need to know the name of the things being contained,
+         * e.g. 'lessons' or 'articles'.
+         */
+        'collection-name': GObject.ParamSpec.string('collection-name',
```

Comment:
>  But what to do about sets that contain, let's say, lessons, articles, videos, and podcasts all together?

Then presumably you would choose a collection-name that encompassed all of those, e.g. "items". That seems like a Content problem rather than an engineering one